course7/course_en.md

```diff
@@ -7,9 +7,11 @@ Imperative programming is a style of programming where we give the computer a se
 However, in the real world, we often need our programs to do more than just calculations. We might need them to read from a file, write to the screen, or change data in memory. These actions are called "side effects," and they can make our programs behave differently each time they run, even with the same input. This breaks the referential transparency, making our programs harder to understand and predict.
 
 ```moonbit
-let x: Int = 1 + 1 // x can be directly replaced with 2
-fn square(x: Int) -> Int { x * x }
-let z: Int = square(x) // can be replaced with { 2 * 2 }, still resulting in 4
+fn init {
+  let x: Int = 1 + 1 // x can be directly replaced with 2
+  fn square(x: Int) -> Int { x * x };
+  let z: Int = square(x) // can be replaced with { 2 * 2 }, still resulting in 4
+}
```

Comment:
Why not `moonbit expr`?

Reply:
This is an init block so it can't be inside the `expr`

Having `expr` is equivalent to
```
fn init {
  <code goes here>
}
```

Reply:
What I mean is that it seems to be better to mark the code block as `moonbit expr` instead of wrapping it with `fn init`.

Reply:
Ok, I see, you are suggesting
```suggestion
```moonbit expr
let x: Int = 1 + 1 // x can be directly replaced with 2
fn square(x: Int) -> Int { x * x };
let z: Int = square(x) // can be replaced with { 2 * 2 }, still resulting in 4
```
Yeah, that looks better